src/main/java/frc/robot/subsystems/Turret.java

```diff
@@ -44,10 +44,10 @@ public Turret() {
    * @param speed the voltage to apply to the motor as a percentage from -1 to 1
    */
   public void set(double speed) {
-    if ((getAngle() < Constants.Turret.MIN_ANGLE && speed > 0.0)
-        || (getAngle() > Constants.Turret.MAX_ANGLE && speed < 0.0)
+    if ((getAngle() < Constants.Turret.MIN_ANGLE && speed < 0.0)
+        || (getAngle() > Constants.Turret.MAX_ANGLE && speed > 0.0)
         || (getAngle() > Constants.Turret.MIN_ANGLE && getAngle() < Constants.Turret.MAX_ANGLE)) {
-      m_turretMotor.set(ControlMode.PercentOutput, speed);
+      m_turretMotor.set(ControlMode.PercentOutput, -speed);
```

Comment:
Please add the direction to the javadoc so this can be better understood. Is +ve speed extending or retracting the hood?